aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -66,20 +69,66 @@ impl TransactionMetadata {
                 // Deprecated. Return an empty vec because we cannot do anything
                 // else here, only `unreachable!` otherwise.
                 TransactionPayload::ModuleBundle(_) => vec![],
+
+                TransactionPayload::Payload(TransactionPayloadInner::V1 {
+                    executable,
+                    extra_config,
+                }) => {
+                    if extra_config.is_multisig() {
+                        vec![]
+                    } else {
+                        match executable {
+                            TransactionExecutable::Script(s) => {
+                                HashValue::sha3_256_of(s.code()).to_vec()
+                            },
+                            _ => vec![],
+                        }
+                    }
+                },
             },
             script_size: match txn.payload() {
-                TransactionPayload::Script(s) => (s.code().len() as u64).into(),
+                TransactionPayload::Script(s)
+                | TransactionPayload::Payload(TransactionPayloadInner::V1 {
+                    executable: TransactionExecutable::Script(s),
+                    ..
+                }) => (s.code().len() as u64).into(),
+
                 _ => NumBytes::zero(),
             },
             is_keyless: aptos_types::keyless::get_authenticators(txn)
                 .map(|res| !res.is_empty())
                 .unwrap_or(false),
             entry_function_payload: match txn.payload() {
```

Comment:
I think the payload needs to be abstract all of these

Reply:
Done.